superset-frontend/plugins/plugin-chart-cartodiagram/src/thematic/components/OlChartMap.tsx

```diff
@@ -623,29 +624,29 @@ export const OlChartMap = (props: OlChartMapProps) => {
         return;
       }
 
-      const val = clickedFeature.get(geomColumn);
+      const val = clickedFeature.get(idColumn);
       if (val === undefined || val === null) {
         return;
       }
 
       // We reset the filter if a filtered feature is clicked again.
-      const resetFilter = filterState.selectedValues?.includes(val);
+      const resetFilter = filterState.selectedValues === val;
 
       setDataMask({
         extraFormData: {
           filters: resetFilter
             ? []
             : [
                 {
-                  col: geomColumn,
-                  op: 'IN' as const,
-                  val: [val],
+                  col: idColumn,
+                  op: '==',
```

Comment:
This should still be an `IN` filter, since this is the one being typically used by cross filters.